x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformHealthIssue.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.transforms;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.common.time.TimeUtils;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class TransformHealthIssue implements Writeable, ToXContentObject {
+
+    public static final String NAME = "data_frame_transform_health_issue";
+
+    private final String issue;
+    private final String details;
+    private final int count;
+    private final Instant firstOccurrence;
+
+    public static final ParseField ISSUE = new ParseField("issue");
+    public static final ParseField DETAILS = new ParseField("details");
+    public static final ParseField COUNT = new ParseField("count");
+    public static final ParseField FIRST_OCCURRENCE = new ParseField("first_occurrence");
+
+    public static final ConstructingObjectParser<TransformHealthIssue, Void> PARSER = new ConstructingObjectParser<>(
+        NAME,
+        true,
+        a -> new TransformHealthIssue((String) a[0], (String) a[1], (Integer) a[2], (Instant) a[3])
+    );
+
+    static {
+        PARSER.declareString(constructorArg(), ISSUE);
+        PARSER.declareString(optionalConstructorArg(), DETAILS);
+        PARSER.declareInt(optionalConstructorArg(), COUNT);
+        PARSER.declareField(
+            optionalConstructorArg(),
+            p -> TimeUtils.parseTimeFieldToInstant(p, FIRST_OCCURRENCE.getPreferredName()),
+            FIRST_OCCURRENCE,
+            ObjectParser.ValueType.VALUE
+        );
+    }
+
+    public static TransformHealthIssue fromXContent(XContentParser parser) throws IOException {
+        return PARSER.parse(parser, null);
+    }
+
+    public TransformHealthIssue(String issue, String details, int count, Instant firstOccurrence) {
+        this.issue = issue;
+        this.details = details;
+        this.count = count;
```

Comment:
There should be validation that the count is not negative, since `WriteVInt` is being used to serialize it.

Maybe we should validate it's not zero either, as it doesn't make sense to report an issue that has occurred zero times.